### PR TITLE
Deploy Prometheus and ServiceMonitors per cluster

### PR DESCRIPTION
**Pull request by metalmatze**
**What this PR does / why we need it**:
So far we only deployed all the things necessary to start Prometheus. With these changes a Prometheus CRD and a CRD for each ServiceMonitor are now deploy with all clusters.

The Prometheus Operator in the monitoring namespace will immediately see that Prometheus CRD and starts scheduling a Prometheus statefulset/pod. Additonally it generates a configmap from all ServiceMonitors for that Prometheus.
That cluster Prometheus then starts scraping its targets right away.

**Which issue(s) this PR fixes** 
Continues #770 

**Special notes for your reviewer**:
:tada:

### CODE REVIEW

File: api/pkg/controller/cluster/pending.go [hunk 1]
```diff
@@ -690,3 +699,93 @@ func (cc *Controller) ensureEtcdCluster(c *kubermaticv1.Cluster) error {
 
 	return nil
 }
+
+func (cc *Controller) ensurePrometheus(c *kubermaticv1.Cluster) error {
+	proms := map[string]func(data *controllerresources.TemplateData, app, masterResourcesPath string) (*prometheusv1.Prometheus, string, error){
+		"prometheus": controllerresources.LoadPrometheusFile,
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/pending.go]: constant

File: api/pkg/controller/cluster/pending.go [hunk 2]
```diff
@@ -690,3 +699,93 @@ func (cc *Controller) ensureEtcdCluster(c *kubermaticv1.Cluster) error {
...
+				_, err := cc.kubermaticClient.MonitoringV1().Prometheuses(c.Status.NamespaceName).Create(generatedPrometheus)
+				if err != nil {
+					return fmt.Errorf("failed to create prometheus resource: %v", err)
+				}
+				return nil
+			}
+			return err
+		}
+		if prometheus.Annotations[lastAppliedConfigAnnotation] != lastApplied {
+			path, err := getPatch(prometheus, generatedPrometheus)
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/pending.go]: patch not path

File: api/pkg/controller/cluster/pending.go [hunk 3]
```diff
@@ -690,3 +699,93 @@ func (cc *Controller) ensureEtcdCluster(c *kubermaticv1.Cluster) error {
...
+}
+
+func (cc *Controller) ensureServiceMonitors(c *kubermaticv1.Cluster) error {
+	sms := map[string]func(data *controllerresources.TemplateData, app, masterResourcesPath string) (*prometheusv1.ServiceMonitor, string, error){
+		"apiserver":          controllerresources.LoadServiceMonitorFile,
+		"controller-manager": controllerresources.LoadServiceMonitorFile,
+		"etcd":               controllerresources.LoadServiceMonitorFile,
+		"kube-state-metrics": controllerresources.LoadServiceMonitorFile,
+		"machine-controller": controllerresources.LoadServiceMonitorFile,
+		"scheduler":          controllerresources.LoadServiceMonitorFile,
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/pending.go]: Constants for all of them

File: api/pkg/controller/cluster/pending.go [hunk 4]
```diff
@@ -690,3 +699,93 @@ func (cc *Controller) ensureEtcdCluster(c *kubermaticv1.Cluster) error {
...
+				_, err := cc.kubermaticClient.MonitoringV1().ServiceMonitors(c.Status.NamespaceName).Create(generatedServiceMonitor)
+				if err != nil {
+					return fmt.Errorf("failed to create service monitor resource: %v", err)
+				}
+				return nil
+			}
+			return err
+		}
+		if serviceMonitor.Annotations[lastApplied] != lastApplied {
+			path, err := getPatch(serviceMonitor, generatedServiceMonitor)
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/pending.go]: patch not path

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: api/pkg/controller/resources/load_files.go [hunk 5]
```diff
@@ -63,6 +65,9 @@ const (
 	//PrometheusServiceAccountName is the name for the Prometheus serviceaccount
 	PrometheusServiceAccountName = "prometheus"
 
+	//PrometheusName is the name for the Prometheus role
```
Comment (mrIncompetent) [inline api/pkg/controller/resources/load_files.go]: wrong comment?

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: api/pkg/controller/resources/load_files.go (hunk 5 above)
Comment (metalmatze) [reply api/pkg/controller/resources/load_files.go]: Yeah. Role at the end is wrong. Copy Paste error :rage:

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**